### PR TITLE
scripts/ctl.hs:  script to quickly create configured iohk-nixops checkouts

### PR DESCRIPTION
**Pull request by deepfire**
Automate experiment creation on staging:
```
staging@cardano-deployer:~$ ./ctl.hs area new devops-51-route53 Nodes
Cloning into 'devops-51-route53'...
remote: Counting objects: 2008, done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 2008 (delta 2), reused 10 (delta 0), pack-reused 1989
Receiving objects: 100% (2008/2008), 1.10 MiB | 1.98 MiB/s, done.
Resolving deltas: 100% (1261/1261), done.
Branch devops-51-route53 set up to track remote branch devops-51-route53 from origin.
Switched to a new branch 'devops-51-route53'

staging@cardano-deployer:~$ cd devops-51-route53

staging@cardano-deployer:~/devops-51-route53$ cat config.yaml
deploymentName: devops-51-route53
nixPath: nixpkgs=https://github.com/NixOS/nixpkgs/archive/05126bc8503a37bfd2fe80867eb5b0bea287c633.tar.gz
nixopsExecutable: nixops
deploymentFiles:
  - deployments/keypairs.nix
  - deployments/cardano-nodes.nix
  - deployments/cardano-nodes-target-aws.nix
  - deployments/cardano-nodes-env-production.nix
```

A mix of deployments can be specified (`Nodes`, `Explorer` etc.) -- the required files will be listed in `config.yaml`.

### CODE REVIEW

**Comment by domenkozar** [comment]
Nice :) I've been meaning to write a task for this. Feedback:

- it needs to allow to pick an environment (defaulting to none) but also we could pick staging or production
- can you combine deployments? Like Nodes + Explorer?
- I'd remove `area` from cli interface

**Comment by deepfire** [comment]
1. was meaning to do this, yes : -)
2. yes -- just listing them is how it works
3. I thought we want to unify everything in a single program, and I wondered if this could be the first step : -)

cc @domenkozar

**Comment by domenkozar** [comment]
3. We'll do that later on, let's save extra typing for now

**Comment by deepfire** [comment]
@domenkozar, done!

1. `area` subcommand is done
2. `--env` and `--tgt` no select proper subsets of the deployment spec Nix files
3. when a deployment has a `*-target-aws.nix`, but no `--env` was provided, the script fails in a clear way

**Comment by deepfire** [comment]
```
[nix-shell:~]$ ./ctl.hs new devops-51-route53 Nodes Timewarp
ctl.hs: user error (--env not specified, but following deployments need it: [Nodes])

[nix-shell:~]$ ./ctl.hs new devops-51-route53 --env Production Nodes Timewarp
ctl.hs: user error (--tgt not specified, but following deployments need it: [Nodes,Timewarp])

[nix-shell:~]$ ./ctl.hs new devops-51-route53 --env Production --tgt AWS Nodes Timewarp
Cloning into 'devops-51-route53'...
remote: Counting objects: 2040, done.
remote: Compressing objects: 100% (42/42), done.
remote: Total 2040 (delta 25), reused 35 (delta 14), pack-reused 1984
Receiving objects: 100% (2040/2040), 1.11 MiB | 0 bytes/s, done.
Resolving deltas: 100% (1280/1280), done.
Branch devops-51-route53 set up to track remote branch devops-51-route53 from origin.
Switched to a new branch 'devops-51-route53'

-- config.yaml is:
deploymentName: devops-51-route53
nixPath: nixpkgs=https://github.com/NixOS/nixpkgs/archive/05126bc8503a37bfd2fe80867eb5b0bea287c633.tar.gz
nixopsExecutable: nixops
deploymentFiles:
  - deployments/keypairs.nix
  - deployments/cardano-nodes.nix
  - deployments/cardano-nodes-env-production.nix
  - deployments/cardano-nodes-target-aws.nix
  - deployments/timewarp.nix
  - deployments/timewarp-target-aws.nix
```

**Comment by domenkozar** [comment]
Rather than fail, I think default env should be `development` and default target should be `aws`.

**Comment by deepfire** [comment]
Ah, I was following your previous recommendation:

> it needs to allow to pick an environment (defaulting to none)

..so I thought it was intended to require explicitness : -)

Fixing..

**Comment by deepfire** [comment]
@domenkozar, this raises another question -- there is no `deployments/cardano-nodes-env-development.nix` -- what would be the policy here?

**Comment by domenkozar** [comment]
If env settings don't exist, don't include them.

**Comment by deepfire** [comment]
But without the
```
let
  nodes = import ./cardano-nodes-config.nix;
  nodeStagConf = import ./../modules/cardano-node-staging.nix;
in {
  resources = {
    elasticIPs = mkNodeIPs nodes accessKeyId;
  };
} // (mkNodes nodes (i: r: nodeStagConf))
```
part, wouldn't the resulting deployment become broken?

**Comment by domenkozar** [comment]
Nope, you don't need to use staging/production

**Comment by deepfire** [comment]
New subcommand: `change`:
```
[nix-shell:~]$ ./ctl.hs new devops-51-route53 Nodes Timewarp
Cloning into 'devops-51-route53'...
remote: Counting objects: 2040, done.
remote: Compressing objects: 100% (44/44), done.
remote: Total 2040 (delta 25), reused 33 (delta 12), pack-reused 1984
Receiving objects: 100% (2040/2040), 1.11 MiB | 0 bytes/s, done.
Resolving deltas: 100% (1280/1280), done.
Branch devops-51-route53 set up to track remote branch devops-51-route53 from origin.
Switched to a new branch 'devops-51-route53'

-- config.yaml is:
deploymentName: devops-51-route53
nixPath: nixpkgs=https://github.com/NixOS/nixpkgs/archive/05126bc8503a37bfd2fe80867eb5b0bea287c633.tar.gz
nixopsExecutable: nixops
deploymentFiles:
  - deployments/keypairs.nix
  - deployments/cardano-nodes.nix
  - deployments/cardano-nodes-target-aws.nix
  - deployments/timewarp.nix
  - deployments/timewarp-target-aws.nix

[nix-shell:~]$ ./ctl.hs change devops-51-route53 --env Production Nodes Timewarp

-- config.yaml updated to:
deploymentName: devops-51-route53
nixPath: nixpkgs=https://github.com/NixOS/nixpkgs/archive/05126bc8503a37bfd2fe80867eb5b0bea287c633.tar.gz
nixopsExecutable: nixops
deploymentFiles:
  - deployments/keypairs.nix
  - deployments/cardano-nodes.nix
  - deployments/cardano-nodes-env-production.nix
  - deployments/cardano-nodes-target-aws.nix
  - deployments/timewarp.nix
  - deployments/timewarp-target-aws.nix

[nix-shell:~]$ ./ctl.hs change devops-51-route53 --env Production Nodes

-- config.yaml updated to:
deploymentName: devops-51-route53
nixPath: nixpkgs=https://github.com/NixOS/nixpkgs/archive/05126bc8503a37bfd2fe80867eb5b0bea287c633.tar.gz
nixopsExecutable: nixops
deploymentFiles:
  - deployments/keypairs.nix
  - deployments/cardano-nodes.nix
  - deployments/cardano-nodes-env-production.nix
  - deployments/cardano-nodes-target-aws.nix
```

**Comment by deepfire** [comment]
@domenkozar, GitHub now requires  a review checkmark, after all these changes..

File: scripts/ctl.hs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  -- subcommand "area" "Operate on an iohk-nixops checkouts"
+  (   subcommand "new"    "Checkout a new iohk-nixops experiment from BRANCH, for a specified set of deployments"
+    (New    <$> branchParser "iohk-nixops branch to check out"                   <*> envParser <*> tgtParser <*> deploymentsParser)
+  <|> subcommand "change" "Change configuration of the specified iohk-nixops experiment"
+    (Change <$> branchParser "iohk-nixops experiment to update configuration of" <*> envParser <*> tgtParser <*> deploymentsParser)
+  )
+
+

+ttl = unsafeTextToLine
```
Comment (domenkozar) [inline scripts/ctl.hs]: This is not used.

File: scripts/ctl.hs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+#! nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/05126bc8503a37bfd2fe80867eb5b0bea287c633.tar.gz
+
+{-# LANGUAGE GADTs, OverloadedStrings, RecordWildCards, UnicodeSyntax #-}
+
+import           Data.Monoid               ((<>))
+import           Data.Maybe
+import qualified Data.Text                         as T
+import           Data.Text                 (Text)
+import           Filesystem.Path.CurrentOS         hiding (concat, empty, null)
+-- import           Prelude.Unicode
```
Comment (domenkozar) [inline scripts/ctl.hs]: Let's remove this line if it's not used

File: scripts/ctl.hs (hunk 1 above)
Comment (deepfire) [inline scripts/ctl.hs]: Done

File: scripts/ctl.hs (hunk 2 above)
Comment (deepfire) [inline scripts/ctl.hs]: Done.

**Comment by domenkozar** [comment]
@deepfire last thing: missing `.travis.yaml` check

**Comment by domenkozar** [comment]
`/home/travis/.travis/job_stages: line 54: ./scripts/ctl.hs: Permission denied`

**Pull request merged**